CMakeLists.txt

```diff
@@ -748,15 +745,51 @@ if(BUSTED_LUA_PRG)
     set_target_properties(functionaltest-lua PROPERTIES FOLDER test)
 endif()
 
-if(LUACHECK_PRG)
-  add_custom_target(lualint
-    COMMAND ${LUACHECK_PRG} -q runtime/ scripts/ src/ test/
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
-else()
-  add_custom_target(lualint false
-    COMMENT "lualint: LUACHECK_PRG not defined")
-endif()
-
+find_program(LUACHECK_PRG luacheck)
+find_program(STYLUA_PRG stylua)
+find_program(SHELLCHECK_PRG shellcheck)
+find_program(FLAKE8_PRG flake8)
+find_program(UNCRUSTIFY_PRG uncrustify)
+
+set(TARGET "lint_lua")
+add_custom_target(${TARGET}
+  COMMAND ${CMAKE_COMMAND}
+  -DPROGRAM=${LUACHECK_PRG}
+  -DPROJECT_ROOT=${PROJECT_SOURCE_DIR}
+  -DTARGET=${TARGET}
+  -P ${PROJECT_SOURCE_DIR}/cmake/lint.cmake)
```

Comment:
cmake is forcing us to be verbose here, nothing we can do about that. But this means `cmake/lint.cmake` gains little. Just put the extra 1-2 lines here, the indirection of `cmake/lint.cmake` isn't justified to save 1-2 lines here.